Tests/XcodeprojTests/FunctionalTests.swift

```diff
@@ -92,6 +92,20 @@ class FunctionalTests: XCTestCase {
 #endif
     }
     
+    func testGenerateXcodeprojWithInvalidModuleNames() {
```

Comment:
I think we can do this using a unit test (instead of a functional one). You can pass an `OutputByteStream` object to record warnings. For reference you can look at `warningStream` property in `Sources/PackageLoading/ModuleMapGenerator.swift`